src/test_l3_dsVideoDevice.c

```diff
@@ -0,0 +1,561 @@
+                                                UT_Control_GetMapString(dsErrorMappingTable, status), HDRCapabilities);
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+}
+
+/**
+ * @brief Get Current Frame Rate Mode
+ *
+ * This test gets Current Frame Rate mode of the device
+ * **Pre-Conditions:** None@n
+ * VideoDevice_init Should be called before calling this test
+ *
+ * **Dependencies:** None@n
+ *
+ * **User Interaction:** @n
+ * User or Automation tool should selected GetFRFMode from list
+ *
+ */
+void dsVideoDevice_dsGetFRFMode()
+{
+    dsError_t status   = dsERR_NONE;
+    int32_t frfMode = 0;
+
+    dsVideoDevice_getHandle();
+    status = dsGetFRFMode(gHandle, &frfMode);
+    UT_LOG_INFO("Result dsGetSupportedVideoCodingFormats(), dsError_t=[%s] FRFMode=[0x08%d]", \
+                                                UT_Control_GetMapString(dsErrorMappingTable, status), frfMode);
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+}
+
+/**
+ * @brief Get Current Frame Rate 
+ *
+ * This test gets Current Frame Rate of the device
+ * **Pre-Conditions:** None@n
+ * VideoDevice_init Should be called before calling this test
+ *
+ * **Dependencies:** None@n
+ *
+ * **User Interaction:** @n
+ * User or Automation tool should selected GetCurrentDisplayframerate from list
+ *
+ */
+void dsVideoDevice_dsGetCurrentDisplayframerate()
+{
+    dsError_t status   = dsERR_NONE;
+    char currentFrameRate[dsVIDEO_FRAMERATE_MAX];
+
+    dsVideoDevice_getHandle();
+    status = dsGetCurrentDisplayframerate(gHandle, currentFrameRate);
+    UT_LOG_INFO("Result dsGetSupportedVideoCodingFormats(), dsError_t=[%s] Current Frame Rate=[%s]", \
+                                                UT_Control_GetMapString(dsErrorMappingTable, status), currentFrameRate);
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+}
+
+/**
+ * @brief Get Current Zoom Mode 
+ *
+ * This test gets Current Zoom Mode of the device
+ * **Pre-Conditions:** None@n
+ * VideoDevice_init Should be called before calling this test
+ *
+ * **Dependencies:** None@n
+ *
+ * **User Interaction:** @n
+ * User or Automation tool should selected GetZoomMode from list
+ *
+ */
+void dsVideoDevice_dsGetZoomMode()
+{
+    dsError_t status   = dsERR_NONE;
+    dsVideoZoom_t dfc;
+
+    dsVideoDevice_getHandle();
+    status = dsGetDFC(gHandle, &dfc);
+    UT_LOG_INFO("Result dsGetDFC(), dsError_t=[%s] Current Frame Rate=[%s]", \
+                                                UT_Control_GetMapString(dsErrorMappingTable, status), \
+                                                UT_Control_GetMapString(dsVideoZoomMappingTable,(int32_t) dfc));
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+}
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+
+int test_l3_dsVideoDevice_register(void)
+{
+    if(gSourceType == 0) {
+        // Create the test suite
+        pSuite = UT_add_suite("[L3 dsVideoDevice - Sink]", NULL, NULL);
```

Comment:
in case of failure must assert , use assert as DS_ASSERT(pSuite != NULL);

Reply:
use UT_add_suite_withGroupID instead of UT_add_suite

Reply:
updated the pull request